spinnaker_camera_driver/cmake/download_spinnaker

```diff
@@ -37,47 +40,140 @@ import glob
 
 logging.basicConfig(level=logging.INFO)
 
-LOGIN_URL = 'https://www.ptgrey.com/login'
-LOGIN_DATA = {
-    'Email': 'code@clearpathrobotics.com',
-    'Password': 'uNjRxoH6NMsJvi6hyPCH'
-    }
+URL_TEMPLATE = URL_TEMPLATES = {
+    'deb': 'https://packages.clearpathrobotics.com/stable/flir/Spinnaker/Ubuntu{version}/spinnaker-1.27.0.48-Ubuntu{version}-{arch}-pkg.tar.gz',
+    'src': None         # if we ever have non-deb archives that require manual extraction, those package URLS will go here.
+}
 
 ARCHS = {
-    'i386': (
-        'https://www.ptgrey.com/support/downloads/11047/',
-        'spinnaker-1.13.0.31-i386',
-        'libSpinnaker.so.1.13.0.31'),
-    'x86_64': (
-        'https://www.ptgrey.com/support/downloads/11048/',
-        'spinnaker-1.13.0.31-amd64',
-        'libSpinnaker.so.1.13.0.31'),
-    'armv7': (
-        'https://www.ptgrey.com/support/downloads/11046/',
-        'spinnaker-1.13.0.31-armhf',
-        'libSpinnaker.so.1.13.0.31'),
-    'armv8': (
-        'https://www.ptgrey.com/support/downloads/11045/',
-        'spinnaker-1.13.0.31-arm64',
-        'libSpinnaker.so.1.13.0.31')
+    'i386': {
+        'linux_arch': 'i386',
+        'current': '16.04',
+        'type': 'deb',
+        'folder_name': 'spinnaker-1.27.0.48-i386',
+        'shared_library': 'usr/lib/libSpinnaker.so.1.27.0.48',
+        'debs': [
```

Comment:
Should we remove this field? it doesn't seem to be used.

Reply:
This was introduced to keep the formatting consistent between the flir + spinnaker drivers & the pointgrey/flycap driver, which has the same build problems & was fixed at the same time.
The flycap driver has some code delivered as .debs and others as pre-compiled but un-packaged archives.
Having extra fields defined here doesn't actively hamper anything and it keeps the formatting consisted across projects.